### PR TITLE
boards: riscv: Convert HiFive1 to DTS

### PR DESCRIPTION
**Pull request by palmer-dabbelt**
Adds DTS bindings for sifive,pwm0, sifive,uart0, sifive,spi0, and
riscv,plic0.

Signed-off-by: Nathaniel Graff <nathaniel.graff@sifive.com>
Signed-off-by: Palmer Dabbelt <palmer@sifive.com>

### CODE REVIEW

**Comment by palmer-dabbelt** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/9195

File: boards/riscv32/hifive1/hifive1.dts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#include <riscv32-fe310.dtsi>
+
+/ {
+	chosen {
+		zephyr,console = &uart0;
+	};
+};
+
+&uart0 {
+	current-speed = <115200>;
```
Comment (galak) [inline boards/riscv32/hifive1/hifive1.dts]: status = "ok";

File: dts/riscv32/riscv32-fe310.dtsi [hunk 2]
```diff
@@ -0,0 +1,178 @@
+/ {
```
Comment (galak) [inline dts/riscv32/riscv32-fe310.dtsi]: Need status properties on the peripheral nodes.  The typical pattern is that peripherals are status = "disabled" by default, and the board.dts explicitly enables. them with a status = "ok".

**Review by galak — CHANGES_REQUESTED** [review]
Some minor dts updates related to use of status property.

File: dts/riscv32/riscv32-fe310.dtsi (hunk 2 above)
Comment (galak) [reply dts/riscv32/riscv32-fe310.dtsi]: No interrupt controllers are typically always enabled, so we don't bother with status on them.  Its more things that have external pin/signals that would be board/application specific.  While interrupts can obviously fall in that group, its almost always the case that at least one internal interrupt ends being needed.

**Review by galak — APPROVED** [review]

**Pull request merged**